app/controllers/discovered_hosts_controller.rb

```diff
@@ -59,24 +59,16 @@ def destroy
   end
 
   def edit
-    unless @host.nil?
-      @host         = @host.becomes(::Host::Managed)
-      @host.type    = 'Host::Managed'
-      @host.managed = true
-      @host.build   = true
-    end
-
+    @host = ::ForemanDiscovery::HostConverter.to_managed(@host) unless @host.nil?
```

Comment:
this does need a transaction, provision from the UI isn't working because of this 
on the plus side your transaction warning is fine :)